pyccel/ast/operators.py

```diff
@@ -559,6 +567,14 @@ def __new__(cls, arg1, arg2, simplify = False):
         if simplify:
             if isinstance(arg2, PyccelUnarySub):
                 return PyccelAdd(arg1, arg2.args[0], simplify = True)
+            elif isinstance(arg1, Literal) and isinstance(arg2, Literal):
+                dtype, precision = cls._calculate_dtype(arg1, arg2)
+                if dtype is NativeInteger():
+                    return LiteralInteger(arg1.python_value - arg2.python_value, precision=precision)
+                if dtype is NativeReal():
+                    return LiteralFloat(arg1.python_value - arg2.python_value, precision=precision)
+                if dtype is NativeComplex():
+                    return LiteralComplex(arg1.python_value - arg2.python_value, precision=precision)
```

Comment:
Codacy is correct that `LiteralComplex` does not have a constructor which takes just a complex. It is up to you whether you add this constructor, or you modify the call here to:
```
result = arg1.python_value - arg2.python_value
return LiteralComplex(result.real, result.imag, precision=precision)
```